sso-kit-client/src/SingleSignOnData.d.ts

```diff
@@ -14,6 +14,6 @@ export type SingleSignOnData = Readonly<{
   authenticated: boolean;
   roles: string[];
   loginLink: string;
-  logoutLink?: string;
+  logoutLink: string;
```

Comment:
The logout link is undefined while the user is not logged in. It only can get value after the login but we get the `SingleSignOnData` from the server even before the login. In that case, based on the server logic, the logoutUrl is not included in the returned json as it is null.

Reply:
I didn't realize that, thanks! I'll revert the property to be optional again.

This also makes me think: what is we have multiple providers defined on the server side? Which login link is used? It's a common use case with social logins, for example.

Reply:
The login link in the returned `SingleSignOnData` is populated from the [SingleSignOnProperties](https://github.com/vaadin/sso-kit/blob/main/sso-kit-starter-hilla/src/main/java/dev/hilla/sso/starter/SingleSignOnContext.java#L121). If it is defined in the `application.properties` then that `login-route` is used, otherwise the default value which is the `/login`. The `/login` on the client side doesn't trigger a login process, it requires a `login` route to be defined or a redirect mechanism to a provider login link. On the `/login` page, if created, the user can create a login link(s) based on the [registeredProviders](https://github.com/vaadin/sso-kit/blob/main/sso-kit-client/src/SingleSignOnContext.ts#L61) which are basically the `registrationIds` of the providers.

The logoutLink is more simple. It is created based on the actually [authorized provider](https://github.com/vaadin/sso-kit/tree/main/sso-kit-clientt).

Reply:
Probably this could be improved with a solution to return all the login links (`/oauth2/authorization/{registrationId}`) to the user in a list and then the user could call them either directly or through a `login()` function with a parameter.

Or to make the `ssoContext.login()` to accept a `registrationId` parameter and to build a login link inside this function. The `registrationIds` are available in the `ssoContext`.

Reply:
This would be a good idea, let's keep it in mind for a future improvement!